pulsar-broker/src/main/java/org/apache/pulsar/broker/rest/TopicsBase.java

```diff
@@ -101,7 +117,11 @@
 @Slf4j
 public class TopicsBase extends PersistentTopicsBase {
 
+    private static final Map<String, Consumer<org.apache.pulsar.client.api.schema.GenericRecord>>
```

Comment:
How is the consumer lifecycle controlled? Does the consumer reside on just a single broker in a Pulsar cluster? Does a consumer get closed automatically when Pulsar load balancing moves the topic to another broker. Does a consumer expire?

Reply:
We will pin the consumer in the specific broker After consumer-created, We will return the consumer broker URL, and the user must use this URL to do the next action.
So, After this, we will not affect by the topic moved.

Reply:
That is not a good model. It will create a very difficult API to use for the user. We need to do it such that it doesn’t matter which broker the consumer will hit.